pyproject.toml

```
@@ -1,6 +1,110 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "altair"
+authors = [ {name = "Vega-Altair Contributors"} ]
+dependencies = [
+    "importlib_metadata; python_version<\"3.8\"",
+    "typing_extensions>=4.0.1; python_version<\"3.11\"",
+    "jinja2",
+    "jsonschema>=3.0",
+    "numpy",
+    "pandas>=0.18",
+    "toolz"
+]
+description = "Vega-Altair: A declarative statistical visualization library for Python."
+readme = "README.md"
+keywords = [
+    "declarative",
+    "statistics",
+    "visualization",
+    "interactive",
+    "json",
+    "vega-lite",
+]
+requires-python = ">=3.7"
+dynamic = ["version"]
+license-files = { paths = ["LICENSE"] }
+classifiers= [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+
+[project.urls]
+Documentation = "https://altair-viz.github.io"
+Source = "https://github.com/altair-viz/altair"
+
+[project.optional-dependencies]
+dev = [
+    "hatch",
+    "ruff",
+    "black<24",
+    "ipython",
+    "pytest",
+    "pytest-cov",
+    "m2r",
+    "vega_datasets",
+    # following is rejected by PyPI
+    "altair_viewer @ git+https://github.com/altair-viz/altair_viewer.git",
+    "vl-convert-python",
+    "mypy",
+    "pandas-stubs",
+    "types-jsonschema",
+    "types-setuptools"
+]
+doc = [
+    "sphinx",
+    "docutils",
+    "jinja2",
+    "numpydoc",
+    "pillow",
+    "pydata-sphinx-theme",
+    "geopandas",
+    "myst-parser",
+    "sphinx_copybutton",
+    "sphinx-design"
+]
+
+[tool.hatch.version]
+path = "altair/__init__.py"
+
+[tool.hatch.metadata]
+allow-direct-references = true
+
+[tool.hatch.build]
+include = [
+    "/altair",
+    "README.md",
+    "LICENSE",
+    "pyproject.toml"
+]
+
+[tool.hatch.envs.default]
+features = ["dev"]
+
+[tool.hatch.envs.default.scripts]
+test = [
+    "black --diff --color --check --line-length 88 .",
```

```suggestion
    "black --diff --color --check .",
```
Specifying the line length should not be necessary as it's the default for black